File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/service/ServiceAsyncTask.java
package com.appcoins.sdk.billing.service;

import android.os.AsyncTask;
import java.util.List;
import java.util.Map;

public class ServiceAsyncTask extends AsyncTask<Object, Object, RequestResponse> {

  private final String httpMethod;
  private final List<String> paths;
  private final Map<String, String> queries;
  private final Map<String, Object> body;
  private final Map<String, String> header;
  private final BdsService bdsService;
  private final String baseUrl;
  private final String endPoint;
  private final ServiceResponseListener serviceResponseListener;

  ServiceAsyncTask(BdsService bdsService, String baseUrl, String endPoint, String httpMethod,
      List<String> paths, Map<String, String> queries, Map<String, String> header,
      Map<String, Object> body, ServiceResponseListener serviceResponseListener) {
    this.bdsService = bdsService;
    this.baseUrl = baseUrl;
    this.endPoint = endPoint;
    this.httpMethod = httpMethod;
    this.paths = paths;
    this.queries = queries;
    this.header = header;
    this.body = body;
    this.serviceResponseListener = serviceResponseListener;
  }

  @Override protected RequestResponse doInBackground(Object[] objects) {
    return bdsService.createRequest(baseUrl, endPoint, httpMethod, paths, queries, header, body);
  }

  @Override protected void onPostExecute(RequestResponse requestResponse) {
    super.onPostExecute(requestResponse);
    if (serviceResponseListener != null) {
      serviceResponseListener.onResponseReceived(requestResponse);
    }
  }
}

File: appcoins-in-game-updates/src/main/java/com/appcoins/sdk/ingameupdates/services/ServiceAsyncTask.kt
package com.appcoins.sdk.ingameupdates.services

import android.os.AsyncTask

class ServiceAsyncTask internal constructor(
    private val bdsService: BdsService,
    private val baseUrl: String,
    private val endPoint: String,
    private val httpMethod: String,
    private val paths: List<String>,
    private val queries: Map<String, String>,
    private val header: Map<String, String>,
    private val body: Map<String, Any>,
    private val serviceResponseListener: ServiceResponseListener?
) : AsyncTask<Any?, Any?, RequestResponse>() {
    override fun doInBackground(objects: Array<Any?>): RequestResponse {
        return bdsService.createRequest(baseUrl, endPoint, httpMethod, paths, queries, header, body)
    }

    override fun onPostExecute(requestResponse: RequestResponse) {
        super.onPostExecute(requestResponse)
        serviceResponseListener?.onResponseReceived(requestResponse)
    }
}
